### DISCUSSION TITLE
Clarification on paramsSerializer option

### DISCUSSION THREAD

**Discussion by zach-mcnulty** [axios/axios#4883] [Q&A] [answered]
Can anyone provide additional information about the updated paramsSerializer option?

I have a fairly simple use case, where I need a query string with duplicate keys converted from an array value in an object, ie someurl.com?arr=1&arr=2. I've seen this same issue scattered around the docs and discussions, with others bemoaning brackets that appear in a query string. 

I am under the impression that setting paramsSerializer to {indexes: null} should correct this issue. Is this correct? If so, what are the reasons that I might be running into errors?

**Comment by DigitalBrainJS** [comment, marked as answer]
> Is this correct?

Yes, this is correct for Axios v1.0.0-alpha.1

```js
  axios.post("https://httpbin.org/post", data, {
    params: {
      foo: "bar",
      arr: [1, 2, 3]
    },
    paramsSerializer: {
      indexes: null
    }
  });
```
Resolved query string:
```
post?foo=bar&arr=1&arr=2&arr=3
```
[Playground](https://codesandbox.io/s/axios-paramsserializer-indexes-ews7rn?file=/src/index.js)

Summary:
```
indexes: null -> post?foo=bar&arr=1&arr=2&arr=3
indexes: false (default) -> post?foo=bar&arr[]=1&arr[]=2&arr[]=3
indexes: true -> post?foo=bar&arr[0]=1&arr[1]=2&arr[2]=3
```

**Reply by zach-mcnulty** [reply]
Ah, so it is not currently available in .27.x. Thank you!
